src/ControlSystem/Systems/Shape.hpp

```diff
@@ -69,8 +69,9 @@ struct Shape : tt::ConformsTo<protocols::ControlSystem> {
     const auto compact_index = iter.compact_index(i);
     if (compact_index.has_value()) {
       iter.set(compact_index.value());
-      return {"(l,m)=("s + get_output(iter.l()) + ","s + get_output(iter.m()) +
-              ")"s};
+      const int m = i < num_components / 2 ? static_cast<int>(iter.m())
+                                           : -static_cast<int>(iter.m());
```

Comment:
I think this should use `iter.coefficient_array()`, but the docs of `SpherepackIterator` should probably be improved to explain this.